CONTRIBUTING.md

```diff
@@ -109,6 +106,66 @@ The contribution process is outlined below:
    maintainer. At this point, a Meta employee will be notified to merge your PR,
    due to tooling limitations.
 
+## Commit Messages
+
+We build Velox for the long-run, and to do so it is crucial that project
+maintainers are able to efficiently inspect and understand project logs.
+
+Commit messages that follow a strict pattern improve maintainability, and allow
+tasks such as summarizing changelogs and identifying API breaking changes to be
+automated. Despite requiring more rigor from authors and reviewers, consistency
+creates productivity in the long term.
+
+In Velox, commit messages are generated based on the input provided to PRs, and
+must follow the [conventional commit
+specification](https://www.conventionalcommits.org/en/v1.0.0/) in the following
+manner:
+
+**PR titles** must follow the pattern:
+
+> \<type\>[optional scope]: \<description\>
+
+where:
+
+* *Type* can be any of the following keywords:
+  * **feat** when new features are being added.
+  * **fix** for bug fixes.
+  * **build** for build or CI-related improvements.
+  * **docs** for enhancements to documentation (only).
+  * **refactor** for refactoring (no logic changes).
+
+* PR titles also take an *optional scope* field containing the area
+  of the code being target by the PR, to further help commit classification.
+  For example, "fix(expression): " or "refactor(parquet): " or "feat(memory):".
+  It is ok to omit this field if there is no adequate classification for the
+  PR, or if the PR touches multiple different components.
+
+* A *description* sentence summarizing the PR, written in imperative tone. The
+  description must be capitalized, not contain a period at the end, and be kept
+  under a 80 characters limit.
+
+Examples of PR titles are:
+
+* feature(type): Add support for IPPREFIX type
+* fix: Prevent unnecessary flatmap to map conversion
+* refactor(vector): Use 'if constexpr' for Buffer::is_pod_like_v\<T\>
+
+The **PR body** must contain a summary of the change, focusing on the *what*
+and *why*. Keep columns at 80 characters.
+
+**Breaking API Changes.** A "BREAKING CHANGE:" footer must be added to PRs
```

Comment:
Should we impose this now? We do not have a fixed public API yet, so users/reviewers will find it hard to determine this.

Reply:
I'm planning to publish that External API docs we discussed a few weeks ago soon so we start socializing this more broadly. I feel like at least across the maintainers and reviewers group we should know what the main external API are (although the codebase organization does not yet reflect that).

Reply:
> A "BREAKING CHANGE:" footer must be added to PRs

Why footer? Should this not be the very first line in the PR description?

Reply:
That's from the original conventional commit spec. An alternative is an exclamation mark after the type `feat(type)!: ....`.